src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -906,7 +906,10 @@ void ComputeRareFlags(NodeFactory factory, bool relocsOnly)
                 flags |= (uint)EETypeRareFlags.RequiresAlign8Flag;
             }
 
-            if (metadataType != null && metadataType.IsHfa)
+            if (metadataType != null &&
```

Comment:
Binder only sets `IsHfa` for ARM:

https://github.com/dotnet/corert/blob/ab5bba191dfbd9b5bd76c701fcaf734b7da4aa1a/src/Native/Runtime/inc/eetype.inl#L524-L529

Reply:
@fadimounir Is this also needed on ARM64?

Reply:
@MichalStrehovsky This is incorrect. The flag is needed on ARM64. The binder doesn't know about ARM64. Example:
https://github.com/dotnet/corert/blob/ab5bba191dfbd9b5bd76c701fcaf734b7da4aa1a/src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs#L323-L328

Reply:
@MichalStrehovsky I would fix this API instead to make it compute the HFA flag only for ARM/ARM64 architectures, so that the IsHfa property would return a more meaningful answer:
https://github.com/dotnet/corert/blob/bd7692c6ab69079fdaa543a0964fc0c1ebb17284/src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs#L493-L503

Reply:
Nope, we want to be able to report HFAness elsewhere. See discussion at https://github.com/dotnet/corert/pull/1310#discussion-diff-64125610. The type might still be considered HFA elsewhere even though we don't mark the EEType as such.

Reply:
Interesting... Yeah in that case we should just control the flags we put on the EEType.